core/src/test/scala/neotypes/CompositeTypesSpec.scala

```diff
@@ -117,6 +117,17 @@ class CompositeTypesSpec extends BaseIntegrationSpec[Future] {
       }
   }
 
+  it should "construct an IncoercibleException message with a field name and value" in execute { s =>
+    final case class PersonIntName(name: Int, born: Int, extra: Option[Int])
+    "match (p:Person {name: 'Charlize Theron'}) return p"
+      .query[PersonIntName]
+      .single(s)
+      .transformWith{
+        case scala.util.Failure(f) => assert(f.getMessage == "Cannot coerce STRING to Java int for field [name] with value [\"Charlize Theron\"]")
+        case scala.util.Success(_) => fail("Query succeeded, but should have failed with IncoercibleException")
+      }
```

Comment:
This is not the correct way to test for a failed future.

Check out other tests, there is a `recoverToExceptionIf[ExClass]` which will transform the `Future[A]` into a `Future[ExClass]` if it failed with that exception or fail if not.
Then you can `map` that exception to check the message.

Note that, if you do not want to check anything about the exception and only that it was thrown you an use `recoverToSucceedIf[ExType]`.